toolchain/semantics/semantics_ir.h

```diff
@@ -38,13 +38,8 @@ class SemanticsIR {
  private:
   friend class SemanticsParseTreeHandler;
 
-  // As noted under cross_reference_irs_, the current IR must always be at
-  // index 1. This is a constant for that.
-  static constexpr auto ThisIR = SemanticsCrossReferenceIRId(1);
-
   explicit SemanticsIR(const SemanticsIR* builtin_ir)
-      : cross_reference_irs_(
-            {builtin_ir == nullptr ? this : builtin_ir, this}) {}
+      : cross_reference_irs_({builtin_ir == nullptr ? this : builtin_ir}) {}
```

Comment:
FWIW, I think this is fine, but also fine with having two private constructors, zero-arg and one-arg, for the two cases if you find that cleaner. Mostly, the private encapsulation of that seems nice, but can then get the internal member initialized the right way.

Reply:
At the root of this, I feel a bit burned by trampling the copy constructor (grrr!). I'm mixed on this current approach because of the single-character difference. But I was thinking that by only providing one constructor, it emphasizes the rule about cross_reference_irs_ construction. Moreso I find the single-character difference in copy versus correct constructor use troublesome (thus, my mulling to _stop_ using constructors for initialization entirely, but still dubious that it's best).

Reply:
And I should add -- _not_ providing the argument-less constructor when I have that single-character difference means, to me, that the uses seem *sightly* less likely to be mistaken for a copy, since obviously nullptr can't be copied.